qml: Even faster blurred Drawer

fredldotme opened the pull request:
The BackgroundBlur component used to be busy on redrawing the blur source whenever the source rectangle changed. This turned out to be the main reason for weaker devices to have problems drawing with the blur shader enabled.

Instead of changing the source rect based on the Drawer's drag distance, do the following:

- Make the BackgroundBlur stay in place attached to the left-hand side.
- Use a parent item with clipping enabled to cause the blur effect to not show up outside of the Drawer's boundaries.
- Change the blur rectangle to only reference the area of interest.

Fixes blurred Drawer performance on more devices
as tested on the JingPad, as well as the weird
jelly effect during drag which was introduced
in the last performance changeset.

fredldotme committed:
qml: Even faster blurred Drawer

The BackgroundBlur component used to be busy on redrawing
the blur source whenever the source rectangle changed.
This turned out to be the main reason for weaker devices
to have problems drawing with the blur shader enabled.

Instead of changing the source rect based on the Drawer's
drag distance, do the following:

- Make the BackgroundBlur stay in place attached to
  the left-hand side.
- Use a parent item with clipping enabled to cause
  the blur effect to not show up outside of the
  Drawer's boundaries.
- Change the blur rectangle to only reference the area
  of interest.

Fixes blurred Drawer performance on more devices
as tested on the JingPad, as well as the weird
jelly effect during drag which was introduced
in the last performance changeset.

maciek134 commented on qml/Launcher/Launcher.qml:
```diff
@@ -366,17 +366,26 @@ FocusScope {
+        height: drawer.height
+        BackgroundBlur {
+            id: backgroundBlur
+            x: 0
```
x/y should default to zero already, maybe this is not needed?

fredldotme replied:
I believe I did it this way to make it rather obvious and explicit how things work rather than expecting eventual readers to understand it immediately.

maciek134 reviewed (approved):
The x/y stuff would be nice to check, but is basically just style, so doesn't matter that much :)

maciek134 commented:
Verified on the Nexus 5 and 4, no noticeable lag, I guess the only slower device could be the BQ4.5? Similar CPU with more L1 cache than N4, but much slower GPU.

maciek134 commented:
CI seems to be failing with `Target 'tests/CMakeFiles/xvfballtests.dir/all' not remade because of errors.`, but no details and no other failed tests.

Flohack74 commented on qml/Launcher/BackgroundBlur.qml:
```diff
@@ -35,10 +35,7 @@ Item {
 
```
Plz add additional copyright at beginning of file

fredldotme replied:
Added

fredldotme committed:
src: Enable Drawer Blur by default

Testing shows that the new changes improve the performance
so much that it is feasible to enable it across the board.

fredldotme committed:
qml: Add copyright to BackgroundBlur
